sda-download/.github/integration/tests/common/70_check_download.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+if [ "$STORAGETYPE" = s3notls ]; then
+    exit 0
+fi
+
+cd dev_utils || exit 1
+
+# get a token, set up variables
+token=$(curl --cacert certs/ca.pem "https://localhost:8000/tokens" | jq -r  '.[0]')
+dataset="https://doi.example/ty009.sfrrss/600.45asasga"
+file="dummy_data"
+expected_size=1048605
+C4GH_PASSPHRASE=$(grep -F passphrase config.yaml | sed -e 's/.* //' -e 's/"//g')
+export C4GH_PASSPHRASE
+
+# download decrypted full file,  check file size
+curl --cacert certs/ca.pem -H "Authorization: Bearer $token" "https://localhost:8443/s3/$dataset/$file" --output full1.bam
+file_size=$(stat -c %s full1.bam)  # Get the size of the file
+
+if [ "$file_size" -ne "$expected_size" ]; then
+    echo "Incorrect file size for full decrypted file"
+    exit 1
+fi
+
+# test that start, end=0 returns the whole file
+curl --cacert certs/ca.pem -H "Authorization: Bearer $token" "https://localhost:8443/s3/$dataset/$file?startCoordinate=0&endCoordinate=0" --output full2.bam
+
+cmp --silent full1.bam full2.bam
+status=$?
+if [[ $status != 0 ]]; then
+    echo "Full decrypted files, with and without coordinates, are different"
+    exit 1
+fi
+
+# download encrypted full file, check that it can be decrypted correctly 
+curl --cacert certs/ca.pem -H "Authorization: Bearer $token" "https://localhost:8443/s3-encrypted/$dataset/$file" --output full3.bam.c4gh
+full_file_size=$(stat -c %s full3.bam.c4gh)
+expected_encrypted_size=1049205
+
+if [ "$full_file_size" -ne "$expected_encrypted_size" ]; then
+    echo "Incorrect file size for full encrypted file"
+    exit 1
+fi
+
+crypt4gh decrypt --sk c4gh.sec.pem < full3.bam.c4gh > full3.bam
+cmp --silent full1.bam full3.bam
+status=$?
+if [[ $status != 0 ]]; then
+    echo "Full encrypted files is not correct when decrypting"
+    exit 1
+fi
+
+# download full encrypted file, test that start, end=0 returns the whole file
+curl --cacert certs/ca.pem -H "Authorization: Bearer $token" "https://localhost:8443/s3-encrypted/$dataset/$file?startCoordinate=0&endCoordinate=0" --output full4.bam.c4gh
```

Comment:
I think it will be good to add these tests also to the file `./dev_utils/run_integration_test.sh` and `dev_utils/run_integration_test_no_tls.sh`

Reply:
As far as I know, the `dev_utils` is to be removed. I'm not sure how the tests should be structured in the future (@jbygdell?)